cmake/ConfigFileSetting.cmake

```diff
@@ -108,7 +108,7 @@ endif()
 if(NRN_ENABLE_MECH_DLL_STYLE)
   set(NRNMECH_DLL_STYLE 1)
 else()
-  set(NRNMECH_DLL_STYLE 0)
+  unset(NRNMECH_DLL_STYLE)
```

Comment:
set defines variable with value 0 and unset undefined variable. I assume you verified that this doesn't change any behaviour.

Reply:
I see that ```src$ grep -r -w NRNMECH_DLL_STYLE *``` indeed has ```#if ...defined(NRNMECH_DLL_STYLE)... for all occurrences of that name.

Reply:
There is actually one single checked

> set defines variable with value 0 and unset undefined variable. I assume you verified that this doesn't change any behaviour.

I checked with autotools and it's like that everywhere, as Michael also pointed out. The thing is in our use of CMake (+spack) we were never setting NRNMECH_DLL_STYLE to 0.

Reply:
ok understood.